File: Introduccion POO/Ej1/Persona.py
class Persona:
    def __init__(self, nombre, edad, ciudad):
        self.nombre = nombre
        self.edad = edad
        self.ciudad = ciudad
# Inciso a
    def mostrar_saludo(self):
        return f"Hola, soy {self.nombre} de {self.ciudad}"
# Inciso c
    def verificar_si_esmayor(self):
        if self.edad >= 18:
            return f"{self.nombre} es mayor de edad"
        else:
            return f"{self.nombre} es menor de edad"
# Inciso b
persona = Persona("Belen", 24, "Brasil")
print(persona.mostrar_saludo())
print(persona.verificar_si_esmayor())
persona2 = Persona("Joel", 23, "Bolivia")
print(persona2.mostrar_saludo())
print(persona2.verificar_si_esmayor())
persona3 = Persona("Tobias", 3, "Alemania")
print(persona3.mostrar_saludo())
print(persona3.verificar_si_esmayor())

File: Introduccion POO/Ej1/Persona.java
package Ej1;
public class Persona {
    public String nombre;
    public int edad;
    public String ciudad;

    public Persona(String nombre, int edad, String ciudad){
        this.nombre = nombre;
        this.edad = edad;
        this.ciudad = ciudad;
    }
    // Inciso a
    public void mostrar_saludo(){
        System.out.println("Hola, soy " + this.nombre + " de " + this.ciudad);
    }
    // Inciso c
    public void verificar_mayoria_edad(){
        if(this.edad >= 18){
            System.out.println(this.nombre + " es mayor de edad");
        }else{
            System.out.println(this.nombre + " es menor de edad");
        }
    }
    // Inciso b
    public static void main(String[] args) {
        Persona p1 = new Persona("Joel", 23, "Bolivia");
        p1.mostrar_saludo();
        p1.verificar_mayoria_edad();
        Persona p2 = new Persona("Belen", 24, "Brasil");
        p2.mostrar_saludo();
        p2.verificar_mayoria_edad();
        Persona p3 = new Persona("Tobias", 2, "Alemania");
        p3.mostrar_saludo();
        p3.verificar_mayoria_edad();
    }
}
